burn-core/src/module/param/constant.rs

```diff
@@ -100,3 +110,79 @@ constant!(i64);
 constant!(i32);
 constant!(i16);
 constant!(i8);
+
+impl<const D: usize, B: Backend> Module<B> for Tensor<B, D> {
+    type Record = ConstantRecord;
+
+    fn visit<V: ModuleVisitor<B>>(&self, visitor: &mut V) {
+        // Important:
+        // We need to implement visit method for Tensor Module because
+        // to_device will be called during the visit method of the ModuleVisitor
+
+        // We are using a dummy param id because the visit method requires a param id
+        let dummy_param_id = ParamId::new();
+        visitor.visit(&dummy_param_id, self)
+    }
+
+    fn map<M: ModuleMapper<B>>(self, mapper: &mut M) -> Self {
+        // Important:
+        // We need to implement visit method for Tensor Module because
+        // to_device will be called during the visit method of the ModuleVisitor
+
+        // We are using a dummy param id because the visit method requires a param id
+        let dummy_param_id = ParamId::new();
+        mapper.map(&dummy_param_id, self)
+    }
```

Comment:
Eventhough it's simpler, I think I would not implement `map` and `visit`, but all other methods manually. This way, it would avoid an allocation for the Param and it's more explicit. For example, I would not add the parameter count to the `num_params`, but I would handle `devices`,  `to_device` and `fork` correctly.

Reply:
We discussed offline and we concluded Module little more work, which will be tracked here: https://github.com/burn-rs/burn/issues/461.